docs/v7-migration.md

```diff
@@ -223,19 +223,19 @@ require('a');
 
 If you were relying on Babel to inject `"use strict"` into all of your CommonJS modules automatically, you'll want to explicitly use the `transform-strict-mode` plugin in your Babel config.
 
-## Separation between the React and Flow presets
+## Separation of the React and Flow presets
 
-`babel-preset-react` has always included the flow plugin automatically from the beginning. This has actually caused a lot of issues with users that accidently use `flow` syntax without intending due to a typo, or adding it in without typechecking with `flow` itself, resulting in errors.
+`babel-preset-react` has always included the flow plugin. This has caused a lot of issues with users that accidently use `flow` syntax unintentionally due to a typo, or adding it in without typechecking with `flow` itself, resulting in errors.
 
-This became further of an issue after we decided to support TypeScript with the help of the TS team. If you wanted to use the react and typescript presets, we would have to figure out a way to turn on/off the syntax automatically via file type or the directive. In the end it seemed easiest to just separate the presets entirely.
+This issue was compounded when we decided to support TypeScript. If you wanted to use the React and TypeScript presets, we would have to figure out a way to turn on/off the syntax, automatically, via file type or the directive. In the end, it was easier to separate the presets entirely.
 
-So now the react preset and the flow preset are separated.
+The React and flow presets are now separate.
 
 ```diff
 {
 -  "presets": ["@babel/preset-react"]
-+  "presets": ["@babel/preset-react", "@babel/preset-flow"] // remove flow types
-+  "presets": ["@babel/preset-react", "@babel/preset-typescript"] // remove typescript types
++  "presets": ["@babel/preset-react", "@babel/preset-flow"] // add flow types
```

Comment:
Thoughts on something like "parse/remove {flow|ts} types" instead?

Reply:
Adding `@babel/preset-flow` will add type checking for flow? Or have I misunderstood?

Seems like it's adding, not removing those types?

Reply:
Common misconception, but babel-preset-flow and babel-preset-typescript both essentially enable our parser to parse their syntax, and then strip it:

https://github.com/babel/babel/blob/master/packages/babel-preset-flow/src/index.js

https://github.com/babel/babel/blob/master/packages/babel-preset-typescript/src/index.js